bindings/go/api/extern.go

```diff
@@ -20,16 +21,28 @@ func SetOnGetBlockHeaderHash(fn func(toBeHashed []byte) []byte) {
 }
 
 // SetOnGetAtv ...
-func SetOnGetAtv(fn func(id []byte) []byte) {
-	ffi.OnGetAtv = fn
+func SetOnGetAtv(fn func(idBytes entities.AtvID) []byte) {
+	ffi.OnGetAtv = func(id []byte) []byte {
+		var idBytes entities.AtvID
+		copy(idBytes[:], id)
```

Comment:
do as little copies as possible

Reply:
Sadly, it's not possible here as we have different sizes for IDs. First I tried to write a similar code in `ffi/extern.go` but the whole difference was is just more boilerplate code